### PR TITLE
build(deps)!: upgrade dependencies to fix vulnerable issues and drop support of node 10, 11, 12, 13, 15

### PR DESCRIPTION
**Pull request by ivan-mattr**
## Description

Upgrade the dependencies which cause vulnerability.

- [ ] Tests for the changes have been added (for bug fixes / features)
- [x] The commit message(s) follow [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Documentation has been added / updated (for bug fixes / features)
- [x] Changes follow the **[contributing](../CONTRIBUTING.md)** document.

## Motivation and Context

To eliminate the vulnerable dependencies.

### After upgrade

Screen Shot 2022-06-20 at 15 57 13

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Which merge strategy will you use?

- [x] Squash
- [ ] Rebase (REVIEW COMMITS)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,31 +52,36 @@
   "devDependencies": {
     "@commitlint/cli": "11.0.0",
```
Comment (ZedMattr) [inline package.json:53]: These two might better also be updated the same way as:
https://github.com/mattrglobal/bbs-signatures/pull/131/files

Also husky might also need to be updated to 8.0

File: .github/workflows/push-binary-dispatch.yml [hunk 2]
```diff
@@ -26,15 +26,12 @@ jobs:
...
+          # exclude windows node 14 due to another node-gyp bug
           - os: windows-latest
-            node-version: 11.x
-          # exclude windows node 15 due to another node-gyp bug
-          - os: windows-latest
-            node-version: 15.x
+            node-version: 14.x
           # exclude windows node 16 due to another node-gyp bug
           - os: windows-latest
             node-version: 16.x
```
Comment (ivan-mattr) [inline .github/workflows/push-binary-dispatch.yml:37]: Hi @tplooker 
Per the conversations in slack, only node 14 and 16 are in scope,  but the jest test has problems to run with node 14 and 16 on Windows, so basically this job `build_win_pub_binaries` is gonna be ignored at all, I guess there is no binary for Windows be published.

File: .github/workflows/push-binary-dispatch.yml [hunk 3]
```diff
@@ -46,7 +43,7 @@ jobs:
       - run: yarn
       - run: yarn install --frozen-lockfile
       - run: yarn build
-      - run: yarn test
+      - run: yarn test # Or shall the test step be skipped?
```
Comment (ivan-mattr) [inline .github/workflows/push-binary-dispatch.yml:46]: @tplooker 
If this `build_win_pub_binaries` job is required, maybe have to skip the test step?

File: .github/workflows/push-binary-dispatch.yml (hunk 2 above)
Comment (ivan-mattr) [reply .github/workflows/push-binary-dispatch.yml]: Per conversation with @tplooker , removed this redundant Windows build job.

File: .github/workflows/push-binary-dispatch.yml [hunk 4]
```diff
@@ -19,46 +19,14 @@ on:
         type: boolean
 
 jobs:
```
Comment (tplooker) [inline .github/workflows/push-binary-dispatch.yml:21]: Can you remove the windows option from the workflow_dispatch option above?

**Review by ZedMattr — APPROVED** [review]

**Comment by ZedMattr** [comment]
Lets add a breaking change commit to signal this will be breaking changes.

**Review by ZedMattr — CHANGES_REQUESTED** [review]
Lets add a breaking change commit to auto upgrade versions.

**Review by ZedMattr — APPROVED** [review]

**Pull request merged**